pcsx2/Recording/InputRecordingFile.cpp

```diff
@@ -158,29 +158,19 @@ bool InputRecordingFile::OpenExisting(const wxString path)
 
 bool InputRecordingFile::ReadKeyBuffer(u8 &result, const uint &frame, const uint port, const uint bufIndex)
 {
-	if (recordingFile == nullptr)
-	{
-		return false;
-	}
-
 	long seek = getRecordingBlockSeekPoint(frame) + controllerInputBytes * port + bufIndex;
-	if (fseek(recordingFile, seek, SEEK_SET) != 0 || fread(&result, 1, 1, recordingFile) != 1)
-	{
-		return false;
-	}
-
-	return true;
+	return fseek(recordingFile, seek, SEEK_SET) == 0 && fread(&result, 1, 1, recordingFile) == 1;
```

Comment:
very nice refactor.  However, we are dropping the `nullptr` check, perhaps that should remain?

Reply:
ReadKeyBuffer is only called when a recording is in Replay mode. With the changes brought on by play-fixes and such, Replay mode should only possible with a file loaded. Therefore, a nullptr check wouldn't be necessary.

Reply:
Fair enough, it just feels a bit fragile depending on those external guarantees.  If those were to ever change, the file handling code can no longer stand on it's own.